packages/node_modules/@webex/internal-plugin-wdm/src/device/device.js

```diff
@@ -255,28 +260,68 @@ const Device = WebexPlugin.extend({
       });
     });
 
+    this.isInNetwork = false;
     this.on('change:serviceHostMap', this._updateServiceCatalog);
 
-    this.listenToAndRun(this, 'change:intranetInactivityCheckUrl', () => this._resetLogoutTimer());
-    this.listenToAndRun(this, 'change:intranetInactivityDuration', () => this._resetLogoutTimer());
-    this.listenTo(this.webex, 'user-activity', () => { this.lastUserActivityDate = Date.now(); });
+    this.listenToAndRun(this, 'change:intranetInactivityCheckUrl', () => { this._resetLogoutTimer(); this._inNetworkResetLogoutTimer(); });
+    this.listenToAndRun(this, 'change:intranetInactivityDuration', () => { this._resetLogoutTimer(); this._inNetworkResetLogoutTimer(); });
+    this.listenToAndRun(this, 'change:inNetworkInactivityDuration', () => { this._resetLogoutTimer(); this._inNetworkResetLogoutTimer(); });
+    this.listenTo(this.webex, 'user-activity', () => {
+      this.lastUserActivityDate = Date.now();
+      this.lastInNetworkUserActivityDate = Date.now();
+    });
+    /**
+   * Meeting started and ended event listenrs
+   * When client enters into the meeting set isInMeeting to true
+   * This tells also triggers both the timers inturn un-setting the timers until meeting ends
+   * On meeting end/clean-up the timers are reset.
+   */
+    this.listenTo(this.webex, 'meeting started', () => {
+      this.isInMeeting = true;
+      this._resetLogoutTimer();
+      this._inNetworkResetLogoutTimer();
+    });
+    this.listenTo(this.webex, 'meeting ended', () => {
+      this.isInMeeting = false;
+      this._resetLogoutTimer();
+      this._inNetworkResetLogoutTimer();
+    });
   },
 
   /**
-   * Don't log the features object
-   * @param {number} depth
-   * @returns {Object}
-   */
+ * Don't log the features object
+ * @param {number} depth
+ * @returns {Object}
+ */
   inspect(depth) {
     return util.inspect(omit(this.serialize(), 'features'), {depth});
   },
 
   isPreDiscoveryService(service) {
-    // The Promise.resolve here is temporary. A future PR will make the
-    // corresponding _ method async to allow for lazy device registration
+  // The Promise.resolve here is temporary. A future PR will make the
+  // corresponding _ method async to allow for lazy device registration
     return Promise.resolve(this._isService(this.config.preDiscoveryServices, service));
   },
 
+  /**
+ * Meeting started and Meeting Ended
+ * These are device.internal functions that can be accessible from web-client
+ * These are used to trigger the events for meeting start/end from within web client.
+ * @returns {boolean}
+ */
+  meetingStarted() {
+    this.webex.trigger('meeting started');
+  },
+  /**
+ * Meeting started and Meeting Ended
+ * These are device.internal functions that can be accessible from web-client
+ * These are used to trigger the events for meeting start/end from within web client.
+ * @returns {boolean}
+ */
+  meetingEnded() {
+    this.webex.trigger('meeting ended');
+  },
```

Comment:
ehhh. I have mixed feelings about putting these on the device and having the client call them to trigger other sdk "things". I realize these will go to plugin-meetings in the future, so it's probably fine

Reply:
These are going to be moved into plugins meetings when its ready. I had a chat with Christopher DeBois he too mentioned we will need to do this then.